### PR TITLE
Feature/checkout form update

### PR DESCRIPTION
**Pull request by Ahmed-Fahmy-Dev**
- date picker added
- radio buttons added
- button box wrapper added
- grid spacing and margin fixed

### CODE REVIEW

File: client/src/features/checkout/components/PaymentForm/PaymentForm.js [hunk 1]
```diff
@@ -1,44 +1,68 @@
...
-            <MenuItem value="visa">
-              <img src="./assets/images/checkoutPageForm/visa.png" alt="visa" />
-            </MenuItem>
-            <MenuItem value="masterCard">
-              <img src="./assets/images/checkoutPageForm/mastercard.png" alt="mastercard" />
-            </MenuItem>
+          <FormLabel id={cardType.name}>{cardType.label}</FormLabel>
+          <Field component={RadioGrouping} aria-labelledby={cardType.name} row name={cardType.name}>
+            <FormControlLabel
+              componentsProps={{ typography: { gutterBottom: 0 } }}
```
Comment (HIMDanny) [inline client/src/features/checkout/components/PaymentForm/PaymentForm.js]:
```suggestion
              componentsProps={{ typography: { gutterBottom: false } }}
```

File: client/src/features/checkout/components/PaymentForm/PaymentForm.js [hunk 2]
```diff
@@ -1,44 +1,68 @@
...
+              value="visa"
+              control={<Radio />}
+              label={<img src="./assets/images/checkoutPageForm/visa.png" alt="visa" />}
+            />
+            <FormControlLabel
+              componentsProps={{ typography: { gutterBottom: 0 } }}
```
Comment (HIMDanny) [inline client/src/features/checkout/components/PaymentForm/PaymentForm.js]:
```suggestion
              componentsProps={{ typography: { gutterBottom: false } }}
```

File: client/src/features/checkout/components/PaymentForm/PaymentForm.js [hunk 3]
```diff
@@ -1,44 +1,68 @@
...
+              value="mastercard"
+              control={<Radio />}
+              label={<img src="./assets/images/checkoutPageForm/mastercard.png" alt="mastercard" />}
+            />
+            <FormControlLabel
+              componentsProps={{ typography: { gutterBottom: 0 } }}
```
Comment (HIMDanny) [inline client/src/features/checkout/components/PaymentForm/PaymentForm.js]:
```suggestion
              componentsProps={{ typography: { gutterBottom: false } }}
```

File: client/src/features/checkout/components/PaymentForm/PaymentForm.js [hunk 4]
```diff
@@ -1,44 +1,68 @@
...
+              value="americanExpress"
+              control={<Radio />}
+              label={<img src="./assets/images/checkoutPageForm/americanExpress.png" alt="americanExpress" />}
+            />
+            <FormControlLabel
+              componentsProps={{ typography: { gutterBottom: 0 } }}
```
Comment (HIMDanny) [inline client/src/features/checkout/components/PaymentForm/PaymentForm.js]:
```suggestion
              componentsProps={{ typography: { gutterBottom: false } }}
```

**Review by HIMDanny — COMMENTED** [review]
Showing error when going to shipping address step:
image

File: client/src/features/checkout/components/PaymentForm/PaymentForm.js (hunk 1 above)
Comment (Ahmed-Fahmy-Dev) [reply client/src/features/checkout/components/PaymentForm/PaymentForm.js]: done

File: client/src/features/checkout/components/PaymentForm/PaymentForm.js (hunk 2 above)
Comment (Ahmed-Fahmy-Dev) [reply client/src/features/checkout/components/PaymentForm/PaymentForm.js]: done

File: client/src/features/checkout/components/PaymentForm/PaymentForm.js (hunk 3 above)
Comment (Ahmed-Fahmy-Dev) [reply client/src/features/checkout/components/PaymentForm/PaymentForm.js]: done

File: client/src/features/checkout/components/PaymentForm/PaymentForm.js (hunk 4 above)
Comment (Ahmed-Fahmy-Dev) [reply client/src/features/checkout/components/PaymentForm/PaymentForm.js]: done

**Comment by AltysS** [comment]
1) I have a general comment. Again, once I left the first input and field validation a received the rest inputs with red errors. 
2) I think it can be good to add validation to the tel number. It can be a regular expression (but it`s optional and up to you) 
3)  The same situation with the payment details as described in the first point. 
4) Please, justify button to the center 
image

**Review by AltysS — APPROVED** [review]

**Review by HIMDanny — APPROVED** [review]

**Pull request merged**